### PR TITLE
 Convert prepare_to_run_t to a run context with information to load/execute the runtime

### PR DESCRIPTION
**Pull request by elinor-fung**

### CODE REVIEW

File: src/corehost/cli/hostpolicy/hostpolicy.cpp [hunk 1]
```diff
@@ -234,40 +74,22 @@ int run_as_lib(
             return StatusCode::Success;
         }
 
-        prepare_to_run_t prep{ hostpolicy_init, args, false /* breadcrumbs_enabled */ };
-
         // Build variables for CoreCLR instantiation
```
Comment (vitek-karas) [inline src/corehost/cli/hostpolicy/hostpolicy.cpp]: Update the comment - while context init does more or less what it says, I don't think it should be mentioned here.

File: src/corehost/cli/hostpolicy/hostpolicy.cpp [hunk 2]
```diff
@@ -278,29 +100,26 @@ int run_as_lib(
     return StatusCode::Success;
 }
 
-int run_as_app(
+int run_host_command(
```
Comment (vitek-karas) [inline src/corehost/cli/hostpolicy/hostpolicy.cpp]: I like this separation - nice!

File: src/corehost/cli/hostpolicy/hostpolicy.cpp [hunk 3]
```diff
@@ -311,30 +130,28 @@ int run_as_app(
...
+int run_as_app(
+    hostpolicy_init_t &hostpolicy_init,
+    const arguments_t &args)
+{
+    run_context_t context {};
+    int rc = context.initialize(g_init, args, true /* enable_breadcrumbs */);
+    if (rc != StatusCode::Success)
+        return rc;
+
     // Verbose logging
```
Comment (vitek-karas) [inline src/corehost/cli/hostpolicy/hostpolicy.cpp]: You could also move the property tracing to the `create_coreclr` as it's done by both cases.

File: src/corehost/cli/hostpolicy/run_context.h [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+#ifndef __RUN_CONTEXT_H__
+#define __RUN_CONTEXT_H__
+
+#include <pal.h>
+
+#include "args.h"
+#include "coreclr.h"
+#include "hostpolicy_init.h"
+
+struct run_context_t
```
Comment (vitek-karas) [inline src/corehost/cli/hostpolicy/run_context.h]: The more I look at this, the less I like this name... "run" is not a good term here.
I would call it "host context", but then we would have a name collision with `hostfxr` which will also have a "host context".
So some ideas:
* `hostpolicy_context` - would make it pretty easy to understand and would nicely differentiate from `hostfxr`. Downside is that we don't really use `hostpolicy_` prefix anywhere.
* `corehost_context` - aligns with the rest of the `corehost_` stuff, but collides with the proposed public `corehost_context_contract` at least a little bit.

I think I would prefer the `hostpolicy_context` - the `corehost_` prefix is really weird, so let's not expand it further (outside of the public APIs).

Comment (elinor-fung) [reply src/corehost/cli/hostpolicy/run_context.h]: Yeah, I renamed this a couple of times and was not really happy with anything...
I prefer `hostpolicy_context` as well - will switch to that.

**Pull request merged**